### PR TITLE
Fix: Resolve issue with skipped execution of sg annotations

### LINKED ISSUE #3581: 默认VPC、默认子网下的kubevirt虚机添加安全组无效
# Bug Report
默认vpc，默认subnet下，kubevirt虚机添加安全组无效
## Expected Behavior
kubevirt虚机添加安全组生效

## Steps to Reproduce the Problem

两种情况：
1. 直接在virt-launcher-i-lra6sgqm-hsv5x Pod的annotation直接添加安全组，是可以生效的。
直接在Pod上添加安全组，这个安全组是禁止ping功能
图片
验证：安全组（禁ping）生效
图片
kube-ovn-controller日志
图片
2. 删除以上的安全组配置，验证ping的通，说明移除了安全组配置
图片
验证ping
图片
kube-ovn-controller日志
图片
3. 以上是通过直接配置launcher Pod，是没有问题的。但是，如果通过配置kubevirt VirtualMachine的annotation添加安全组（4-7步骤），编辑后重启，virt-launcher的pod会继承安全组配置，但是这时候安全组配置不生效。
4.  配置kubevirt VirtualMachine
图片
5. 配置后，重启虚机，查看launcher pod，已经同步了安全组配置
图片
6. 验证安全组（禁ping），但是ping的通，说明安全组没有生效
图片
7. 查看kube-ovn-controller日志，似乎没有添加安全组的动作：
图片

## Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

  ```bash
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:58:53Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}

  ```

- kube-ovn version:
  
  ```bash
1.12.4

  ```

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
  **Output of `uname -r`:**

  ```bash
"NFS Server 4.0 (G193)"
 4.19.113-3.nfs.x86_64
  ```

### PR DESCRIPTION
**Pull request by wfnuser**
The problem causing ineffective application of sg annotations is that, during virtual machine restart, the logical switch port is intentionally not deleted.(I guess).

When sg annotations are added and the VM is restarted, the create logical switch port logic is skipped as it detects the existing lsp. Consequently, the annotation fails to attach to the lsp. Even when we sync lsp for sg, it has no effect.

A simple fix is to update the existing lsp during lsp creation if it already exists. This approach ensures correct annotation attachment and addresses the skipped execution issue.

# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Bug fixes

## Which issue(s) this PR fixes

Fixes #3581

### CODE REVIEW

**Comment by wfnuser** [comment]
It's a draft pr.
1) I'm not sure if we should actually delete the pod and related lsp before create news during the restart of vm.
2) If so, we might need to update more fields if the lsp exists.

File: pkg/ovs/ovn-nb-logical_switch_port.go [hunk 1]
```diff
@@ -25,6 +25,14 @@ func (c *OVNNbClient) CreateLogicalSwitchPort(lsName, lspName, ip, mac, podName,
 
 	// ignore
 	if exist {
+		if portSecurity && len(securityGroups) != 0 {
+			lsp, err := c.GetLogicalSwitchPort(lspName, true)
```
Comment (bobz965) [inline pkg/ovs/ovn-nb-logical_switch_port.go:29]: please log the err  if err != nil, and then set the SG

**Review by bobz965 — CHANGES_REQUESTED** [review]

**Comment by bobz965** [comment]
> It's a draft pr.
> 
> 1. I'm not sure if we should actually delete the pod and related lsp before create news during the restart of vm.
> 2. If so, we might need to update more fields if the lsp exists.

1. VM restart should only delete the pod. keep the LSP is betther.

File: pkg/ovs/ovn-nb-logical_switch_port.go (hunk 1 above)
Comment (wfnuser) [reply pkg/ovs/ovn-nb-logical_switch_port.go:29]: updated.

File: pkg/ovs/ovn-nb-logical_switch_port.go [hunk 2]
```diff
@@ -25,6 +25,21 @@ func (c *OVNNbClient) CreateLogicalSwitchPort(lsName, lspName, ip, mac, podName,
...
+			if err != nil {
+				klog.Error(err)
```
Comment (a180285) [inline pkg/ovs/ovn-nb-logical_switch_port.go:31]: log more info, E.g. what are you doing, and then err happened. Like following logging info

Comment (a180285) [reply pkg/ovs/ovn-nb-logical_switch_port.go:31]: And then return error

File: pkg/ovs/ovn-nb-logical_switch_port.go [hunk 3]
```diff
@@ -25,6 +25,21 @@ func (c *OVNNbClient) CreateLogicalSwitchPort(lsName, lspName, ip, mac, podName,
...
+			} else {
+				sgList := strings.Split(securityGroups, ",")
+				if _, err := c.SetLogicalSwitchPortSecurityGroup(lsp, "add", sgList...); err != nil {
+					klog.Errorf("set logical switch port %s security groups %s: %v", lsp.Name, securityGroups, err)
```
Comment (a180285) [inline pkg/ovs/ovn-nb-logical_switch_port.go:35]: return err

File: pkg/ovs/ovn-nb-logical_switch_port.go (hunk 2 above)
Comment (wfnuser) [reply pkg/ovs/ovn-nb-logical_switch_port.go]: The code has been updated to another style.

File: pkg/ovs/ovn-nb-logical_switch_port.go (hunk 3 above)
Comment (wfnuser) [reply pkg/ovs/ovn-nb-logical_switch_port.go]: The code has been updated to another style.

File: pkg/ovs/ovn-nb-logical_switch_port.go [hunk 4]
```diff
@@ -88,7 +77,29 @@ func (c *OVNNbClient) CreateLogicalSwitchPort(lsName, lspName, ip, mac, podName,
...
+		klog.Error(err)
+		return err
+	}
+
+	// update if exists
+	if exist {
+		lsp := buildLogicalSwitchPort(lsName, lspName, ip, mac, podName, namespace, portSecurity, securityGroups, vips, enableDHCP, dhcpOptions, vpc)
+		if err := c.UpdateLogicalSwitchPort(lsp, &lsp.PortSecurity, &lsp.ExternalIDs); err != nil {
+			klog.Error(err)
+			return fmt.Errorf("logical switch port: %s exists and update: %v", lspName, err)
```
Comment (bobz965) [inline pkg/ovs/ovn-nb-logical_switch_port.go:95]:
```suggestion
			return fmt.Errorf("failed to update logical switch port %s: %v", lspName, err)
```

**Comment by oilbeater** [comment]
There are lint issues should be fixed

**Comment by wfnuser** [comment]
> There are lint issues should be fixed

Happy Chinese new year! I just returned to my apartment in BJ. Updated.

**Review by bobz965 — APPROVED** [review]

**Review by oilbeater — APPROVED** [review]

**Comment by oilbeater** [comment]
@wfnuser Thanks!

**Pull request merged**